collections/sort_by.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * Returns all elements in the given collection, sorted by their result using the given selector
+ *
+ * Example:
+ *
+ * ```typescript
+ * const people = [
+ *     { name: 'Anna', age: 34 },
+ *     { name: 'Kim', age: 42 },
+ *     { name: 'John', age: 23 },
+ * ]
+ * const sortedByAge = sortBy(people, it => it.age)
```

Comment:
We set up doc example type checking in #1050 . So this jsdoc example causes an error in CI saying "sortBy is not found". Please add `import { sortBy } from "./sort_by.ts";` at the beginning of the example block.

Reply:
Oh the examples are not evaluated in the scope of the current file? I was not aware. That makes the examples kind of verbose doesn't it?

Reply:
I added the import - but it still feels like a lot of clutter for an example that is meant to be displayed in an LSP hover :-S